### DISCUSSION TITLE
Referencing blueprint objects

### DISCUSSION THREAD

**Discussion by dagillespie** [nxrighthere/UnrealCLR#100] [Help] [answered]
Hi @nxrighthere - have started to dig into UnrealCLR a little more now (great work!) and am looking at current best practice linking BP to C# and calling/referencing functions. On this two points:

- Is there a way to implement World.GetActor<MyBPClass>("optionalName")?  It seems that even if you specify <Actor> as class (of which MyBPClass derives), it still results in a null? Current workaround is using tags World.GetActorByTag("tagName")

-  Also is there any reason why actor.GetText() is present but Actor.GetString() isn't right now? Is that an active design choice or is it on the backlog?

**Comment by nxrighthere** [comment, marked as answer]
> Is there a way to implement World.GetActor("optionalName")? It seems that even if you specify as class (of which MyBPClass derives), it still results in a null? Current workaround is using tags World.GetActorByTag("tagName")

You have to use the real name of the actor, but not the label from the editor to do this.

> Also is there any reason why actor.GetText() is present but Actor.GetString() isn't right now? Is that an active design choice or is it on the backlog?

Blueprints have only the text data type from what I see in the editor and native API.

**Reply by dagillespie** [reply]
Thanks for explaining the GetActorByName. I assume this name therefore isn't consistent each time the applications run as typically the full actor name can change depending on what else is the the level (eg if an AActor is spawned at runtime its name may be an incremental from the last AActor of same class).

Also maybe I'm misunderstanding however UE4 C++ API has FString, FText and FName which need to be converted depending on context. Not sure if there are any limitations by implementing one specific text class over another here for UnrealCLR

**Reply by nxrighthere** [reply]
> Also maybe I'm misunderstanding however UE4 C++ API has FString, FText and FName which need to be converted depending on context. Not sure if there are any limitations by implementing one specific text class over another here for UnrealCLR

Once https://github.com/nxrighthere/UnrealCLR/issues/61 will be resolved, it will be possible to use these data types too. In blueprints, you can only use `FText` from what I see.
